automation_tools/__init__.py

```diff
@@ -1655,16 +1655,168 @@ def setenforce(mode):
     run('setenforce {0}'.format(mode))
 
 
-def host_pings(host, timeout=15):
+# =============================================================================
+# Satellite and Capsule Upgrade
+# =============================================================================
+def copy_ssh_key(from_host, to_host):
+    """This will generate(if not already) ssh-key on from_host
+    and copy that ssh-key to to_host. Beware that both hosts
+    should have authorized key added for test running host.
+
+    :param from_host: A string. Hostname on which the key to be generated and
+        to be copied from.
+    :param to_host: A string. Hostname on to which the ssh-key will be copied.
+
+    """
+    ssh_out = execute(lambda: run('ls .ssh/'), host=from_host)
+    if 'id_rsa.pub' not in ssh_out[from_host]:
+        execute(
+            lambda: run('ssh-keygen -f .ssh/id_rsa -t rsa -N \'\''),
+            host=from_host)
+    local('scp root@{0}:.ssh/id_rsa.pub .'.format(from_host))
+    local('cat id_rsa.pub | ssh -o \'StrictHostKeyChecking no\' '
```

Comment:
> Because I want to copy the contents in authorized_keys file on remote server. And not want to copy the entire file. And believe that this is good way to do that.

OK. Makes sense.

Here's what I mean about getting rid of back slashes:

```python
local("cat id_rsa.pub | ssh -o 'StrictHostKeyChecking no' "
      "root@{0} 'cat >> .ssh/authorized_keys'".format(to_host))
```